x/market/keeper/key.go

```diff
@@ -19,18 +19,30 @@ var (
 func orderKey(id types.OrderID) []byte {
 	buf := bytes.NewBuffer(orderPrefix)
 	buf.Write(id.Owner.Bytes())
-	binary.Write(buf, binary.BigEndian, id.DSeq)
-	binary.Write(buf, binary.BigEndian, id.GSeq)
-	binary.Write(buf, binary.BigEndian, id.OSeq)
+	if err := binary.Write(buf, binary.BigEndian, id.DSeq); err != nil {
```

Comment:
again all of these key generation things should blow up if they fail.  even if we return an error, it's not clear how to handle it gracefully.

Reply:
`panic`ing now